app/domain/operations/event_logs/create.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require "dry/monads"
+require "dry/monads/do"
+
+module Operations
+  module EventLogs
+    # Create Event Log entity
+    class Create
+      send(:include, Dry::Monads[:result, :do])
+
+      attr_accessor :resource_class
+
+      # @param [Hash] opts Options to build audit log event
+      # @option opts [<GlobalID>] :subject_gid required
+      # @option opts [<String>]   :correlation_id required
+      # @option opts [<Symbol>]   :event_category required
+      # @option opts [<String>]   :session_id optional
+      # @option opts [<String>]   :account_id optional
+      # @option opts [<String>]   :host_id required
+      # @option opts [<String>]   :trigger required
+      # @option opts [<String>]   :response required
+      # @option opts [<Symbol>]   :log_level optional
+      # @option opts [<Symbol>]   :severity optional
+      # @option opts [<DateTime>] :event_time required
+      # @option opts [<Array>]    :tags optional
+      # @return [Dry::Monad] result
+      def call(params)
+        values = yield validate(params)
+        entity = yield create(values)
+
+        Success(entity)
+      end
+
+      private
+
+      def validate(params)
+        @resource_class = params.delete(:resource_class)
+        return Failure("resource class name missing") unless resource_class
+
+        result = domain_contract_class.new.call(params)
+        result.success? ? Success(result) : Failure(result)
+      end
+
+      def create(values)
+        Success(
+          domain_entity_class.new(values.to_h)
+        )
+      end
+
+      def domain_contract_class
+        Object.const_get("AcaEntities::EventLogs::#{resource_class}EventLogContract") if defined?("AcaEntities::EventLogs::#{resource_class}EventLogContract")
```

Comment:
this change is not fixing the security incident that we raised, it's just re-writing in a way that codesweep will not complain, but it's the same: https://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-constantize

This could lead to an attacker try to see log that doesn't belong to them or logs of a higher role he is not supposed to do.

Please go and review the two options given; Factory Pattern, or Switch statement that @saikumar9 raised.